#!/usr/bin/env python3

PACKAGE = "env_gazebo"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

# 雷达云台控制（两轴）
gen.add("lidar_gimbal_yaw", double_t, 0, "雷达云台偏航角 (degrees)", 0.0, -180.0, 180.0)
gen.add("lidar_gimbal_pitch", double_t, 0, "雷达云台俯仰角 (degrees)", 90.0, -90.0, 90.0)

# 相机云台控制（单轴俯仰）
gen.add("camera_gimbal_pitch", double_t, 0, "相机云台俯仰角 (degrees)", -20.0, -90.0, 90.0)

# 机器人位置控制
gen.add("robot_x", double_t, 0, "机器人X位置", 0.0, -10.0, 10.0)
gen.add("robot_y", double_t, 0, "机器人Y位置", 0.0, -10.0, 10.0)
gen.add("robot_z", double_t, 0, "机器人Z位置", 3.0, 0.0, 10.0)
gen.add("robot_yaw", double_t, 0, "机器人偏航角 (degrees)", 0.0, -180.0, 180.0)

exit(gen.generate(PACKAGE, "env_gazebo", "GimbalControl"))
